File: OPENAPI-REST-API/swagger-client/java-vertx/src/main/java/io/swagger/server/api/model/HomeServicesVps.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.server.api.model.HomeServicesVpsLinks;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * HomeServicesVps
 */




public class HomeServicesVps   {
  private HomeServicesVpsLinks links = null;

  private Integer count = null;

  public HomeServicesVps links(HomeServicesVpsLinks links) {
    this.links = links;
    return this;
  }

  /**
   * Get links
   * @return links
   **/
    public HomeServicesVpsLinks getLinks() {
    return links;
  }

  public void setLinks(HomeServicesVpsLinks links) {
    this.links = links;
  }

  public HomeServicesVps count(Integer count) {
    this.count = count;
    return this;
  }

  /**
   * Number of VPS services.
   * @return count
   **/
    public Integer getCount() {
    return count;
  }

  public void setCount(Integer count) {
    this.count = count;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    HomeServicesVps homeServicesVps = (HomeServicesVps) o;
    return Objects.equals(this.links, homeServicesVps.links) &&
        Objects.equals(this.count, homeServicesVps.count);
  }

  @Override
  public int hashCode() {
    return Objects.hash(links, count);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class HomeServicesVps {\n");
    
    sb.append("    links: ").append(toIndentedString(links)).append("\n");
    sb.append("    count: ").append(toIndentedString(count)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/HomeServicesVps.cpp


#include "Home_services_vps.h"

using namespace Tiny;

Home_services_vps::Home_services_vps()
{
	links = HomeServicesVpsLinks();
	count = int(0);
}

Home_services_vps::Home_services_vps(std::string jsonString)
{
	this->fromJson(jsonString);
}

Home_services_vps::~Home_services_vps()
{

}

void
Home_services_vps::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *linksKey = "links";

    if(object.has_key(linksKey))
    {
        bourne::json value = object[linksKey];




        HomeServicesVpsLinks* obj = &links;
		obj->fromJson(value.dump());

    }

    const char *countKey = "count";

    if(object.has_key(countKey))
    {
        bourne::json value = object[countKey];



        jsonToValue(&count, value, "int");


    }


}

bourne::json
Home_services_vps::toJson()
{
    bourne::json object = bourne::json::object();






	object["links"] = getLinks().toJson();





    object["count"] = getCount();



    return object;

}

HomeServicesVpsLinks
Home_services_vps::getLinks()
{
	return links;
}

void
Home_services_vps::setLinks(HomeServicesVpsLinks links)
{
	this->links = links;
}

int
Home_services_vps::getCount()
{
	return count;
}

void
Home_services_vps::setCount(int count)
{
	this->count = count;
}



